homeassistant/helpers/event.py

```diff
@@ -1609,7 +1609,15 @@ def _interval_listener(self, now: datetime) -> None:
             self._track_job,
             hass.loop.time() + self.seconds,
         )
-        hass.async_run_hass_job(self._run_job, now)
+        if self._run_job.job_type is HassJobType.Coroutinefunction:
+            hass.async_create_periodic_task(
+                # mypy does not know we just checked for the job type
+                self.action(now),  # type: ignore[arg-type]
+                f"track time interval {self.seconds}",
+                eager_start=True,
+            )
+        else:
```

Comment:
This won't take care of the case where its an executor job. That expands the scope too much for this PR so something to address in the future.

Reply:
https://github.com/home-assistant/core/pull/112640#discussion_r1516917640